test/helpers/kubectl.go

```diff
@@ -2955,20 +2983,19 @@ func (kub *Kubectl) ciliumPreFlightCheck() error {
 // Cilium are in a good state. If one of the multiple preflight fails it'll
 // return an error.
 func (kub *Kubectl) CiliumPreFlightCheck() error {
-	ginkgoext.By("Performing Cilium preflight check")
+	ginkgoext.By("Validating Cilium Installation")
 	// Doing this withTimeout because the Status can be ready, but the other
 	// nodes cannot be show up yet, and the cilium-health can fail as a false positive.
 	var (
-		lastError           string
+		lastError           error
 		consecutiveFailures int
 	)
 
 	body := func() bool {
-		if err := kub.ciliumPreFlightCheck(); err != nil {
-			newError := err.Error()
-			if lastError != newError || consecutiveFailures >= 5 {
-				ginkgoext.GinkgoPrint("Cilium is not ready yet: %s", newError)
-				lastError = newError
+		if err := kub.validateCilium(); err != nil {
+			if lastError.Error() != err.Error() || consecutiveFailures >= 5 {
```

Comment:
This panics if we get any error because `lastError` is initialised with `nil`. Initialising `lastError` with empty error should work here.